### PR TITLE
Add basic support for tracking allocation size

### PR DESCRIPTION
**Pull request by michalbiesek**
~~Only last commit is valid~~
- [x] Required #556
- [x] Required #573 
- [x] Tests ~~required decorators on memkind_memtier layer~~

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/567)

### CODE REVIEW

**Review by PatKamin — COMMENTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/567)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jschmieg)

**Review by bratpiorka — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, 3 of 3 files at r3, 1 of 1 files at r4, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/567)* status: all files reviewed, 1 unresolved discussion (waiting on @jschmieg and @michalbiesek)

___
*[test/memkind_memtier_test.cpp, line 188 at r5](https://reviewable.io/reviews/memkind/memkind/567#-MWNr5f41FEmoxko0xXj:-MWNr5f58KfOpwU_TS4V:bx5o2qh) ([raw file](https://github.com/memkind/memkind/blob/73c8a9c3dd578bb0e6c81954b12335c4b9049f6c/test/memkind_memtier_test.cpp#L188)):*
> ```cpp
>     size_t ptr_usable_size = memtier_usable_size(ptr);
> 
>     ASSERT_EQ(ptr_usable_size, allocation_sum());
> ```

Shouldn't you check here for just "512"? Currently you are using your feature to test itself. 
Or add additional check for allocation_sum() == 512. This applies also to other checks in this test case.

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/567)* status: 3 of 4 files reviewed, 1 unresolved discussion (waiting on @bratpiorka and @jschmieg)

___
*[test/memkind_memtier_test.cpp, line 188 at r5](https://reviewable.io/reviews/memkind/memkind/567#-MWNr5f41FEmoxko0xXj:-MWNuMW2DMXo2D6s1dH4:b-o7simc) ([raw file](https://github.com/memkind/memkind/blob/73c8a9c3dd578bb0e6c81954b12335c4b9049f6c/test/memkind_memtier_test.cpp#L188)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

Shouldn't you check here for just "512"? Currently you are using your feature to test itself. 
Or add additional check for allocation_sum() == 512. This applies also to other checks in this test case.
</blockquote></details>

Done, added the numeric value check

**Review by bratpiorka — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r6.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/567#-:-MWO2RvBB5GTZQX6CIwu:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jschmieg)

File: man/memkind_memtier.3 [hunk 1]
```diff
@@ -53,6 +53,8 @@ functionality is considered as stable API (STANDARD API).
 .BI "size_t memtier_usable_size(void " "*ptr" );
 .br
 .BI "void memtier_free(void " "*ptr" );
+.br
+.BI "size_t memtier_tier_allocated_size(struct memtier_tier  " "*tier");
```
Comment (PatKamin) [inline man/memkind_memtier.3:57]: nit: Remove extra whitespace after `memtier_tier` and add a whitespace before closing brackets.

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/567)* status: 0 of 4 files reviewed, all discussions resolved (waiting on @bratpiorka and @jschmieg)

___
*[man/memkind_memtier.3, line 57 at r3](https://reviewable.io/reviews/memkind/memkind/567#-MW96NvVsJo0n4Laud2K-r3-57:-MWOqzz93xd18LdG-Tm6:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/d5814d6858e50a036920b41c8cea39dcdff9d6b0/man/memkind_memtier.3#L57)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

nit: Remove extra whitespace after `memtier_tier` and add a whitespace before closing brackets.
</blockquote></details>

Done.

**Review by PatKamin — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, 1 of 3 files at r3, 2 of 4 files at r7, 2 of 3 files at r8, 1 of 2 files at r12, 1 of 1 files at r13.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/567#-:-MWOtfUKDRUdtGifbBZq:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @jschmieg and @michalbiesek)

**Review by michalbiesek — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/567)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jschmieg)

___
*[test/memkind_memtier_test.cpp, line 178 at r6](https://reviewable.io/reviews/memkind/memkind/567#-MW96NvVsJo0n4Laud2L-r6-178:-MWOtcvp0hVg9EfJV6w9:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/742b0c7fb28522365fe5a7dd1b0b2e1f121a5b72/test/memkind_memtier_test.cpp#L178)):*
<details><summary><i>Previously, PatKamin (Patryk Kaminski) wrote…</i></summary><blockquote>

nit: 0ULL?
</blockquote></details>

Done.

**Pull request merged**